src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -4103,6 +4103,44 @@ void MacroAssembler::fill_words(Register base, Register cnt, Register value)
   bind(fini);
 }
 
+// Zero blocks of memory by using CBO.ZERO.
+//
+// Aligns the base address first sufficiently for CBO.ZERO, then uses
+// CBO.ZERO repeatedly for every full block.  cnt is the size to be
+// zeroed in HeapWords.  Returns the count of words left to be zeroed
+// in cnt.
+//
+// NOTE: This is intended to be used in the zero_blocks() stub.  If
+// you want to use it elsewhere, note that cnt must be >= CacheLineSize.
+void MacroAssembler::zero_dcache_blocks(Register base, Register cnt, Register tmp1, Register tmp2, Register tmp3) {
+  Label initial_table_end, loop_cbo_zero;
+
+  // Align base with cache line size.
+  neg(tmp1, base);
+  andi(tmp1, tmp1, CacheLineSize - 1);
+
+  // tmp1: the number of bytes to be filled to align the base with cache line size.
+  add(base, base, tmp1);
+  srai(tmp2, tmp1, 3);
+  sub(cnt, cnt, tmp2);
+  add(tmp3, zr, zr);
+  movptr(tmp3, initial_table_end);
+  srli(tmp2, tmp1, 1);
+  sub(tmp2, tmp3, tmp2);
+  j(tmp2);
+  for (int i = -CacheLineSize + 8; i < 0; i += 8) {
+    sd(zr, Address(base, i));
+  }
+  bind(initial_table_end);
+
+  li(tmp1, CacheLineSize >> 3);
```

Comment:
I think it will be more consistent to use 'mv' here instead of 'li'. I am considering making 'li' a private method.